### DISCUSSION TITLE
Access collection item by slug (or any unique field)?

### DISCUSSION THREAD

**Discussion by revnelson** [payloadcms/payload#4724] [Q&A]
Adding a field with `unique: true` creates an Index on the collection. Is there any way to access collection items by unique indexes other than _id_? For example, fetching `https://example.com/api/pages/home` would be much better DX than

```
const paginatedPages = await fetch("https://example.com/api/pages?where[slug][equals]=home").then(async res => await res.json())

if ((paginatedPages?.docs?.length ?? 0) > 0) {
   const page = paginatedPages.docs.at(0)
}
```

**Comment by ChrisGV04** [comment]
Not yet, unfortunately. It would be nice but I think it could have a lot of weird edge-cases or make certain queries a bit slower since you can have multiple `unique` fields inside the same collection, so it would have to check for all of them.

For example: A `User` document could have a unique `id`, `slug` and `email` fields at the same time, so calling `GET /api/users/my-slug` would need to check if an ID, a slug or an email match `my-slug`. It might be inefficient, but I'm not sure.

However, you always can create your own Express route to do exactly that by calling `GET /api/page-by-slug/home` . Yes, it's more work but that's the beauty of Payload's flexibility to add something if you need it even if it doesn't have it currently.
